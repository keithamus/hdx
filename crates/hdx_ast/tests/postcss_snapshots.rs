mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
