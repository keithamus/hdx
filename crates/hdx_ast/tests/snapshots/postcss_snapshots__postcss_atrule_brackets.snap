---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 0,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 9,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 10,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "type": "Function",
                  "name": {
                    "kind": "Function",
                    "offset": 11,
                    "len": 10
                  },
                  "values": [
                    {
                      "kind": "String",
                      "offset": 21,
                      "len": 11
                    },
                    {
                      "kind": "Comma",
                      "offset": 32,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 33,
                      "len": 1
                    },
                    {
                      "type": "SimpleBlock",
                      "start": 34,
                      "open": {
                        "kind": "LeftCurly",
                        "len": 1
                      },
                      "values": {
                        "values": [
                          {
                            "kind": "Whitespace",
                            "offset": 35,
                            "len": 1
                          },
                          {
                            "kind": "String",
                            "offset": 36,
                            "len": 10
                          },
                          {
                            "kind": "Colon",
                            "offset": 46,
                            "len": 1
                          },
                          {
                            "kind": "Whitespace",
                            "offset": 47,
                            "len": 1
                          },
                          {
                            "kind": "Number",
                            "offset": 48,
                            "len": 3
                          }
                        ]
                      },
                      "close": {
                        "kind": "RightCurly",
                        "len": 1
                      }
                    }
                  ],
                  "close_paren": {
                    "kind": "RightParen",
                    "offset": 53,
                    "len": 1
                  }
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 55,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 56,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 57,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 58,
                  "len": 2
                },
                {
                  "type": "SimpleBlock",
                  "start": 60,
                  "open": {
                    "kind": "LeftSquare",
                    "len": 1
                  },
                  "values": {
                    "values": [
                      {
                        "kind": "Ident",
                        "offset": 61,
                        "len": 6
                      }
                    ]
                  },
                  "close": {
                    "kind": "RightSquare",
                    "len": 1
                  }
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 68,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 69,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 75,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 85,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 86,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 87,
                        "len": 11
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 98,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 102,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 104,
          "len": 1
        }
      }
    },
    {
      "type": "stylerule",
      "selectors": [],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 134,
        "len": 1
      },
      "declarations": [],
      "rules": [
        {
          "type": "stylerule",
          "selectors": [
            {
              "components": [
                {
                  "type": "wildcard",
                  "value": {
                    "kind": "Delim",
                    "offset": 138,
                    "len": 1
                  }
                }
              ],
              "comma": null
            }
          ],
          "type": "style-declaration",
          "open": {
            "kind": "LeftCurly",
            "offset": 140,
            "len": 1
          },
          "declarations": [
            {
              "type": "property",
              "name": {
                "kind": "Ident",
                "offset": 142,
                "len": 10
              },
              "colon": {
                "kind": "Colon",
                "offset": 152,
                "len": 1
              },
              "value": {
                "values": [
                  {
                    "kind": "Whitespace",
                    "offset": 153,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 154,
                    "len": 3
                  }
                ]
              },
              "important": null,
              "semicolon": {
                "kind": "Semicolon",
                "offset": 157,
                "len": 1
              }
            }
          ],
          "rules": [],
          "close": {
            "kind": "RightCurly",
            "offset": 159,
            "len": 1
          }
        }
      ],
      "close": {
        "kind": "RightCurly",
        "offset": 161,
        "len": 1
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 164,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 173,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 174,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "type": "SimpleBlock",
                  "start": 175,
                  "open": {
                    "kind": "LeftParen",
                    "len": 1
                  },
                  "values": {
                    "values": [
                      {
                        "kind": "String",
                        "offset": 176,
                        "len": 9
                      },
                      {
                        "kind": "Colon",
                        "offset": 185,
                        "len": 1
                      },
                      {
                        "kind": "Whitespace",
                        "offset": 186,
                        "len": 1
                      },
                      {
                        "kind": "Number",
                        "offset": 187,
                        "len": 1
                      }
                    ]
                  },
                  "close": {
                    "kind": "RightParen",
                    "len": 1
                  }
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 190,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 191,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 192,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 193,
                  "len": 2
                },
                {
                  "kind": "Delim",
                  "offset": 195,
                  "len": 1
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 196,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 197,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 199,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 209,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 210,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 211,
                        "len": 3
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 214,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 216,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 218,
          "len": 1
        }
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 221,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 230,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 231,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "type": "Function",
                  "name": {
                    "kind": "Function",
                    "offset": 232,
                    "len": 5
                  },
                  "values": [
                    {
                      "kind": "String",
                      "offset": 237,
                      "len": 9
                    },
                    {
                      "kind": "Colon",
                      "offset": 246,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 247,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 248,
                      "len": 1
                    }
                  ],
                  "close_paren": {
                    "kind": "RightParen",
                    "offset": 249,
                    "len": 1
                  }
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 251,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 252,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 253,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 254,
                  "len": 2
                },
                {
                  "kind": "Delim",
                  "offset": 256,
                  "len": 1
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 257,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 258,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 260,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 270,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 271,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 272,
                        "len": 3
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 275,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 277,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 279,
          "len": 1
        }
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 282,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 291,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 292,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "kind": "Ident",
                  "offset": 293,
                  "len": 5
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 299,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 300,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 301,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 302,
                  "len": 2
                },
                {
                  "kind": "Delim",
                  "offset": 304,
                  "len": 1
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 305,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 306,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 308,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 318,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 319,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 320,
                        "len": 3
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 323,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 325,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 327,
          "len": 1
        }
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 330,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 339,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 340,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "type": "SimpleBlock",
                  "start": 341,
                  "open": {
                    "kind": "LeftSquare",
                    "len": 1
                  },
                  "values": {
                    "values": []
                  },
                  "close": {
                    "kind": "RightSquare",
                    "len": 1
                  }
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 344,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 345,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 346,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 347,
                  "len": 2
                },
                {
                  "kind": "Delim",
                  "offset": 349,
                  "len": 1
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 350,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 351,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 353,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 363,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 364,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 365,
                        "len": 3
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 368,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 370,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 372,
          "len": 1
        }
      }
    },
    {
      "type": "AtRule",
      "name": {
        "kind": "AtKeyword",
        "offset": 375,
        "len": 9
      },
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 384,
            "len": 1
          },
          {
            "type": "SimpleBlock",
            "start": 385,
            "open": {
              "kind": "LeftParen",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "type": "SimpleBlock",
                  "start": 386,
                  "open": {
                    "kind": "LeftSquare",
                    "len": 1
                  },
                  "values": {
                    "values": [
                      {
                        "kind": "Ident",
                        "offset": 387,
                        "len": 5
                      },
                      {
                        "kind": "Colon",
                        "offset": 392,
                        "len": 1
                      },
                      {
                        "kind": "Whitespace",
                        "offset": 393,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 394,
                        "len": 3
                      }
                    ]
                  },
                  "close": {
                    "kind": "RightSquare",
                    "len": 1
                  }
                }
              ]
            },
            "close": {
              "kind": "RightParen",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 399,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 400,
          "len": 1
        },
        "declarations": [],
        "rules": [
          {
            "type": "QualifiedRule",
            "type": "QualifiedRule",
            "prelude": {
              "values": [
                {
                  "kind": "Whitespace",
                  "offset": 401,
                  "len": 1
                },
                {
                  "kind": "Whitespace",
                  "offset": 402,
                  "len": 2
                },
                {
                  "kind": "Delim",
                  "offset": 404,
                  "len": 1
                }
              ]
            },
            "block": {
              "type": "Block",
              "start": 405,
              "open_curly": {
                "kind": "LeftCurly",
                "offset": 406,
                "len": 1
              },
              "declarations": [
                {
                  "type": "Declaration",
                  "name": {
                    "kind": "Ident",
                    "offset": 408,
                    "len": 10
                  },
                  "colon": {
                    "kind": "Colon",
                    "offset": 418,
                    "len": 1
                  },
                  "value": {
                    "values": [
                      {
                        "kind": "Whitespace",
                        "offset": 419,
                        "len": 1
                      },
                      {
                        "kind": "Ident",
                        "offset": 420,
                        "len": 3
                      }
                    ]
                  },
                  "important": null,
                  "semicolon": {
                    "kind": "Semicolon",
                    "offset": 423,
                    "len": 1
                  }
                }
              ],
              "rules": [],
              "close_curly": {
                "kind": "RightCurly",
                "offset": 425,
                "len": 1
              }
            }
          }
        ],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 427,
          "len": 1
        }
      }
    },
    {
      "type": "stylerule",
      "selectors": [],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 479,
        "len": 1
      },
      "declarations": [],
      "rules": [
        {
          "type": "stylerule",
          "selectors": [
            {
              "components": [
                {
                  "type": "wildcard",
                  "value": {
                    "kind": "Delim",
                    "offset": 483,
                    "len": 1
                  }
                }
              ],
              "comma": null
            }
          ],
          "type": "style-declaration",
          "open": {
            "kind": "LeftCurly",
            "offset": 485,
            "len": 1
          },
          "declarations": [
            {
              "type": "property",
              "name": {
                "kind": "Ident",
                "offset": 487,
                "len": 10
              },
              "colon": {
                "kind": "Colon",
                "offset": 497,
                "len": 1
              },
              "value": {
                "values": [
                  {
                    "kind": "Whitespace",
                    "offset": 498,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 499,
                    "len": 3
                  }
                ]
              },
              "important": null,
              "semicolon": {
                "kind": "Semicolon",
                "offset": 502,
                "len": 1
              }
            }
          ],
          "rules": [],
          "close": {
            "kind": "RightCurly",
            "offset": 504,
            "len": 1
          }
        }
      ],
      "close": {
        "kind": "RightCurly",
        "offset": 506,
        "len": 1
      }
    }
  ]
}
